lib/angular-http-server.js

```diff
@@ -102,24 +105,64 @@ function requestListener(req, res) {
 
     const safeFullFileName = getFilePathFromUrl(req.url, basePath, { baseHref });
 
-    fs.stat(safeFullFileName, function(err, stats) {
-        var fileBuffer;
+    fs.stat(safeFullFileName, function (err, stats) {
         if (!err && stats.isFile()) {
-            fileBuffer = fs.readFileSync(safeFullFileName);
-            let ct = mime.lookup(safeFullFileName);
+            const ct = mime.lookup(safeFullFileName);
+            const acceptEncoding = req.headers['accept-encoding'] || '';
+
             log(`Sending ${safeFullFileName} with Content-Type ${ct}`);
-            res.writeHead(200, { "Content-Type": ct });
+
+            if (contentIsCompressible(ct, acceptEncoding)) {
+                sendCompressedResponse(res, ct, acceptEncoding, safeFullFileName);
+            } else {
+                const raw = fs.createReadStream(safeFullFileName);
+                res.writeHead(200, {
+                    "Content-Type": ct
+                });
+                raw.pipe(res);
+            }
         } else {
             log("Route %s, replacing with index.html", safeFullFileName);
-            fileBuffer = returnDistFile();
             res.writeHead(200, { "Content-Type": "text/html" });
+            res.write(returnDistFile());
+            res.end();
         }
-
-        res.write(fileBuffer);
-        res.end();
     });
 }
 
+function contentIsCompressible(contentType, acceptEncoding) {
+    // Note: This is not a conformant accept-encoding parser.
+    // See https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.3
+    if (COMPRESSIBLE_CONTENT_TYPES.includes(contentType)
+        && (
+            /\bdeflate\b/.test(acceptEncoding) ||
+            /\bgzip\b/.test(acceptEncoding)
+        )) {
+        return true;
+    }
+    return false;
+}
+
+function sendCompressedResponse(res, contentType, encoding, safeFullFileName) {
+    const raw = fs.createReadStream(safeFullFileName);
+
+    if (/\bgzip\b/.test(encoding)) {
+        res.writeHead(200, {
+            "Content-Encoding": "gzip",
+            "Content-Type": contentType
+        });
+        raw.pipe(zlib.createGzip()).pipe(res);
+    } else if (/\bdeflate\b/.test(encoding)) {
+        res.writeHead(200, {
+            "Content-Type": "deflate",
+            "Content-Type": contentType
+        });
+        raw.pipe(zlib.createDeflate()).pipe(res);
+    } else {
+        log("Unsupported encoding: ", encoding);
```

Comment:
if we hit this will a server request being going unanswered?

Reply:
@simonh1000, that is my read.

If you'd like, I can pick this up and carry it to the finish line, including handling unsupported encodings.

Reply:
I'd first like to understand why this feature is useful for people on this development-only server.

Reply:
That's a fair question. I'm not entirely sure myself.

Reply:
It looks like @oderayi's use case was Lighthouse auditing before deploying to production.

Reply:
hmm. auditing like this does not help you know what score you will get in production. I'm concerned that this is increasing the sense that this is a production server, which it absolutely is not. I'm inclined to drop this.